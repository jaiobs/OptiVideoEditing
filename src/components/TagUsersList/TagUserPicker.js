import React, {Component} from 'react';
import {connect} from 'react-redux';
import {
  View,
  StyleSheet,
  Image,
  Text,
  TextInput,
  FlatList,
  TouchableOpacity,
  Platform
} from 'react-native';
import {getTagUsers} from '../../actions/TagUsersListAction';
import Modal from 'react-native-modal';
import {Images} from '../../res';
import FastImage from '@stevenmasini/react-native-fast-image';
import AsyncStorage from '@react-native-async-storage/async-storage';

export class TagCell extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { item, onSelect } = this.props;

    return (
      <TouchableOpacity
        style={[styles.trackItemContainer]}
        onPress={() => {onSelect()}}>
        <View style={{flexDirection: 'row', paddingTop: 5, paddingBottom: 5, paddingLeft:10, flex: 1}}>
          <FastImage
            style={{height: 50, width: 50, borderRadius: 25}}
            source={item.profile_pic != null && item.profile_pic != "" ? {uri: item.profile_pic} : Images.user_placeholder}
            //resizeMode={FastImage.resizeMode.contain}
          />
          
        <View style={{flexDirection: 'column', paddingLeft:5, justifyContent:"center",flex:1, width:"100%"}}>
            <Text style={styles.title}>{item.first_name} {item.last_name}</Text>
            <Text style={[styles.title,{fontSize:14}]}>@{item.user_name}</Text>
            <View style={{backgroundColor: 'rgba(255,255,255,0.6)', height: 1, width: '100%',marginTop:10}} />
        </View>
        </View>
      </TouchableOpacity>
    );
  }
}

class TagUserPicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tagUsers: [],
      URL_BASE:"",
      AUTH_TOKEN:"",
    };
  }

  componentDidMount() {
    //get the tag users
    AsyncStorage.getItem(`BASE_URL`).then((value) => {
        AsyncStorage.getItem(`AUTH_TOKEN`).then((auth_val) => {
          if (value !== null && auth_val !== null) {
                this.setState({
                URL_BASE: value,
                AUTH_TOKEN: auth_val
                },()=>{
                    console.log("TAG USER LIST HIT",value,auth_val)
                    this.props.getTagUsers(value,auth_val,"");
                })
            }
        })
    })
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    //getting tag users
    if (
      nextProps.tagList.rows !== undefined &&
      nextProps.tagList.rows.length > 0 &&
      nextProps.tagList.rows !== this.props.tagList.rows
    ) {
      console.log("TAG PROPS INSIDE")
      this.setState({tagUsers: nextProps.tagList.rows});
    }
  }
  onSelectedItems(item, index) {
    this.props.onUserSelected(item)
  }

  renderTagItem(item,index) {
    return (
      <TagCell item={item} onSelect={() => this.onSelectedItems(item, index)} />
    );
  }

  //search songs
  searchUsers(query) {
      if(query.length > 2){
        this.props.getTagUsers(this.state.URL_BASE,this.state.AUTH_TOKEN,query);
      }    
  }

  render() {
    return (
      <Modal
        //backdropColor={"transparent"} 
        isVisible={this.props.visible}
        onRequestClose={() => {
          this.props.onClosePressed();
        }}
        style={[styles.container]}>

        {/* SEARCH INPUT */}
        <View style={[styles.searchContainer]}>
            {/* CLOSE BUTTON */}
            <TouchableOpacity
                style={{alignSelf:"flex-start",margin:10,marginTop:15}}
                onPress={() => {
                  this.props.onClosePressed();
                }}>
                    <Image style={styles.closeIcon} source={Images.close_icon} />
            </TouchableOpacity>
          <View style={[
            styles.searchInputBox,
            ]}>
                <Image style={styles.searchIcon} source={Images.search_icon}/>
                <TextInput
                placeholder={"Search users"}
                placeholderTextColor={"white"}
                style={[styles.searchInput]}
                onChangeText={query => {
                    this.searchUsers(query);
                }}
                />
          </View>
        </View>

        {/* SEARCH LIST */}
        <FlatList
          style={[styles.trackListStyle]}
          data={this.state.tagUsers}
          bounces={false}
          renderItem={({item,index}) => this.renderTagItem(item,index)}
          contentContainerStyle={{alignSelf:"center",width:"100%"}}
          keyExtractor={item => item.id.toString()}
        />
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-start',
    flexDirection: 'column',
    margin: 0,
    backgroundColor:"#00000060",
    alignItems: 'center',
  },
  searchContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    marginTop: 30,
    alignItems: 'center',
    marginLeft: 5,
    marginRight: 15,
    top:Platform.OS=='ios'?20:0
  },
  title: {color: 'white', marginTop: 5, fontSize:16, flex:1, width:"100%"},
  searchInputBox: {
    flex: 1,
    flexDirection:"row",
    backgroundColor:"#00000090",
    marginLeft: 5,
    marginRight: 5,
  },
  trackListStyle: {
    width: '100%',
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 16,
    marginRight: 30,
  },
  searchIcon: {height:25,width:25,resizeMode:"contain",alignSelf:"center",margin:5},
  searchInput: {color: 'white', height: 40,flex:1},
  trackTitle: {color: 'white', marginLeft: 8, width: '60%'},
  closeIcon: {height: 24, width: 24,resizeMode:'contain'},
  trackItemContainer: {flexDirection: 'column', justifyContent:"center", margin:10},
});

const mapStateToProps = state => {
  const {tagList} = state.TagUsersListReducer;
  return {
    tagList,
  };
};

export default connect(
  mapStateToProps,
  {getTagUsers},
)(TagUserPicker);