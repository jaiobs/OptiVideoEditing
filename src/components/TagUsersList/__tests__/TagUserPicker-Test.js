import React from 'react';
import renderer from 'react-test-renderer';
import TagUserPicker, { TagCell } from '../TagUserPicker';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.useFakeTimers();

const mockStore = configureMockStore({});

const mockItem = {
    profile_pic: 'test-image.pnge',
    first_name: 'firstName',
    last_name: 'lastName',
    user_name: 'userName',
  }
  
const mockNullItem = {
    profile_pic: null,
    first_name: null,
    last_name: null,
    user_name: null,
}

describe("TagUserPicker Tests", () => {
    jest.useFakeTimers();
    it("should render TagUserPicker correctly", async() => {
        
        const tree = renderer
        .create(jsx())
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
})

describe('TagUserPicker render flatlist with defined values', () => {
    jest.useFakeTimers();
    it('should match the snapshot', async() => {
      // will generate snapshot for your component
      const onSelectPress = jest.fn();
      const tree = renderer
        .create(<TagCell item={mockItem} onSelect={onSelectPress} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('TagUserPicker render flatlist with null values', () => {
    jest.useFakeTimers();
    it('should match the snapshot', async() => {
        // will generate snapshot for your component
        const onSelectPress = jest.fn();
        const tree = renderer
          .create(<TagCell item={mockNullItem} onSelect={onSelectPress} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toMatchSnapshot();
      });
  });
  

const jsx = () => {
    const closePressHandler = jest.fn();
    const initialState = {
        TagUsersListReducer: {tagList: mockItem},
      };
    const store = mockStore(initialState);
    return (
        <Provider store={store}>
        <TagUserPicker
          visible={true}
          onClosePressed={closePressHandler}
        />
        </Provider>
    );
};