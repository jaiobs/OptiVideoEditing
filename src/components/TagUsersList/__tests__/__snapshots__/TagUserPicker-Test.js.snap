// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagUserPicker Tests should render TagUserPicker correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000060",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginLeft": 5,
            "marginRight": 15,
            "marginTop": 30,
            "top": 20,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "flex-start",
            "margin": 10,
            "marginTop": 15,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/res/assets/images/cross.png",
            }
          }
          style={
            Object {
              "height": 24,
              "resizeMode": "contain",
              "width": 24,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#00000090",
              "flex": 1,
              "flexDirection": "row",
              "marginLeft": 5,
              "marginRight": 5,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/res/assets/images/search-icon.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 25,
              "margin": 5,
              "resizeMode": "contain",
              "width": 25,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Search users"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "white",
                "flex": 1,
                "height": 40,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <RCTScrollView
      bounces={false}
      contentContainerStyle={
        Object {
          "alignSelf": "center",
          "width": "100%",
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "marginBottom": 10,
            "marginLeft": 16,
            "marginRight": 30,
            "marginTop": 10,
            "width": "100%",
          },
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</Modal>
`;

exports[`TagUserPicker render flatlist with defined values should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingTop": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "test-image.pnge",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 5,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
        firstName
         
        lastName
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 16,
              "marginTop": 5,
              "width": "100%",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        @
        userName
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.6)",
            "height": 1,
            "marginTop": 10,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TagUserPicker render flatlist with defined values should match the snapshot 2`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingTop": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "test-image.pnge",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 5,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
        firstName
         
        lastName
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 16,
              "marginTop": 5,
              "width": "100%",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        @
        userName
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.6)",
            "height": 1,
            "marginTop": 10,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TagUserPicker render flatlist with null values should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingTop": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../src/res/assets/images/user_placeholder.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 5,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
         
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 16,
              "marginTop": 5,
              "width": "100%",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        @
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.6)",
            "height": 1,
            "marginTop": 10,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TagUserPicker render flatlist with null values should match the snapshot 2`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingTop": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../src/res/assets/images/user_placeholder.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 5,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "marginTop": 5,
            "width": "100%",
          }
        }
      >
         
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 16,
              "marginTop": 5,
              "width": "100%",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        @
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.6)",
            "height": 1,
            "marginTop": 10,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;
